docker-files-for-Gitlab-CI-rust/substrate-linux-ci/Dockerfile

```diff
@@ -0,0 +1,42 @@
+# https://github.com/rust-lang-nursery/docker-rust-nightly/blob/master/nightly-slim/Dockerfile
+FROM rustlang/rust:nightly-slim
+
+# metadata
+LABEL maintainer="devops-team@parity.io" \
+	vendor="Parity Technologies" \
+	name="parity/rust-substrate-build" \
+	description="Image for Substrate CI builds and tests." \
+	url="https://github.com/paritytech/scripts/" \
+	vcs-url="scripts/docker-files-for-Gitlab-CI-rust/substrate-linux-ci"
+
+WORKDIR /builds/parity/substrate/
+
+# install tools and dependencies
+RUN apt-get -y update; \
+	apt-get install -y --no-install-recommends \
+		software-properties-common curl git \
+		make cmake g++ clang rhash \
+		pkg-config libudev-dev time \
+		libssl-dev; \
+# apt clean up
+	apt-get autoremove -y; \
+	apt-get clean; \
+	rm -rf /var/lib/apt/lists/*; \
+# setup default stable channel
+	rustup default stable; \
+	cargo install cargo-audit sccache; \
+# install wasm toolchain for polkadot
+	rustup target add wasm32-unknown-unknown --toolchain nightly; \
+# Install wasm-gc. It's useful for stripping slimming down wasm binaries.
+# (polkadot)
+	cargo +nightly install --git https://github.com/alexcrichton/wasm-gc;
+
+# compiler ENV
+ENV CC=gcc \
+	CXX=g++ \
+# cache handlers
+	RUSTC_WRAPPER=sccache \
+	CARGO_HOME=/cargo/substrate/cargo \
+	SCCACHE_DIR=/cargo/substrate/sccache \
```

Comment:
why is CARGO_HOME not within the SCCACHE_DIR?

Reply:
there's no sense in adding it there, these are different caches

Reply:
how? isn't the sccache set up to cache CARGO_HOME? That's the directory where the build fragments reside, no?

Reply:
nope, `sccache` has nothing to do with `CARGO_HOME`.
Sccache cares about compilation cache only.
While `CARGO_HOME` saves dependencies, their git repositories and some other stuff not related to compilation.
That's why I decided to keep both.
Also, `cargo` doesn't bother about concurrent access to `CARGO_HOME` so I had to divide cargo caches by CI job names to avoid data race conditions.

Reply:
so `sccaches` *compilation cache* would not be stored on disk somewhere without it?